wtdbg2.pl

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Getopt::Std;
+
+my %opts = (t=>4, m=>"2g", X=>50, M=>"map-ont");
+getopts('t:x:o:a:g:X:M:OP', \%opts);
+$opts{e} = lc $opts{e};
+die (qq/Usage: wtdbg2.pl [options] <reads.fa>
+Options:
+  -o STR     output prefix [input]
+  -t INT     number of threads [$opts{t}]
+  -x STR     preset: rs, ont, sq []
+  -g NUM     estimated genome size []
+  -X FLOAT   coverage cutoff [$opts{X}]
+  -a STR     additional wtdbg2 options []
+  -M STR     minimap2 preset [$opts{M}]
+  -m NUM     samtools sort block size [$opts{m}]
+  -P         skip minimap2-based polishing
+  -O         output without execution
+/) if (@ARGV == 0);
+
+$opts{asm} = gwhich("wtdbg2") || die;
+$opts{cns} = gwhich("wtpoa-cns") || die;
+$opts{smt} = gwhich("samtools") || die;
+$opts{mm2} = gwhich("minimap2") || die;
+
+my $prefix = defined($opts{o})? $opts{o} : $ARGV[0];
+my $smt_threads = $opts{t} < 4? $opts{t} : 4;
+
+my $asm_opt = "-t $opts{t} -fo $prefix";
+$asm_opt .= " -g $opts{g}" if defined($opts{g}) && $opts{g} =~ /^\d/;
+$asm_opt .= " -x $opts{x}" if defined($opts{x});
+$asm_opt .= " -i $ARGV[$_]" for (0 .. @ARGV-1);
+
+my $mm2_input = join(' ', @ARGV);
+
+my @lines = ();
+push(@lines, qq/$opts{asm} $asm_opt/);
+push(@lines, qq/$opts{cns} -t $opts{t} -i $prefix.ctg.lay.gz -fo $prefix.raw.fa/);
+unless (defined $opts{P}) {
+	push(@lines, qq/$opts{mm2} -a$opts{M} -x $opts{M} -t$opts{t} -r2k $prefix.raw.fa $mm2_input | $opts{smt} sort -m $opts{m} -\@$smt_threads -o $prefix.bam/);
+	push(@lines, qq/$opts{smt} view -F0x900 $prefix.bam | $opts{cns} -t $opts{t} -d $prefix.raw.fa -i - -fo $prefix.cns.fa/);
+}
+
+if (defined $opts{O}) {
+	print "$_\n" for (@lines);
+} else {
+	system($_) for (@lines);
```

Comment:
Might be a good idea to check if each command succeeds before running the next?
eg. `system($_)==0 or die "$_ exited with non-zero code $?"